good-enough-practices-for-scientific-computing.tex

```diff
@@ -434,32 +449,52 @@ \subsection*{Acknowledgments}
 
 \item
   \practice{2b}{Decompose programs into functions} that are no more than
-  one page (about 60 lines) long, do not
-  use global variables (constants are OK), and take no more than half
-  a dozen parameters.  The key motivation here is to fit the program
+  one page (about 60 lines) long. Functions are sections within a program
+  that do one particular thing. The syntax varies depending on programming
+  language, but generally you name the function, list the parameters, or 
+  information needed for that section of code, and list what information will 
+  come out of the function. 
+  % check definition of functions
+  % WOULD BE GOOD TO HAVE EXAMPLE OF A FUNCTION
+  Functions should take no more than about 5 or 6 input parameters
```

Comment:
This may be confusing, if you don't understand 'scope'. Functions actually *do* access data, by it being passed to them, right?

Not sure I have a solution to this.